aiplatform/src/test/java/aiplatform/CreateBatchPredictionGeminiBigqueryJobSampleTest.java

```diff
@@ -0,0 +1,109 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class CreateBatchPredictionGeminiBigqueryJobSampleTest {
+  private static final String PROJECT = System.getenv("UCAIP_PROJECT_ID");
+  private static final String MODEL_ID = "gemini-1.5-flash-002";
+  private static final String BIGQUERY_SOURCE_URI =
+      "bq://storage-samples.generative_ai.batch_requests_for_multimodal_input";
+  private static final String BIGQUERY_DESTINATION_OUTPUT_URI_PREFIX = "bq://ucaip-sample-tests";
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  private PrintStream originalPrintStream;
+  private String batchPredictionJobId;
+
+  private static void requireEnvVar(String varName) {
+    String errorMessage =
+        String.format("Environment variable '%s' is required to perform these tests.", varName);
+    assertNotNull(errorMessage, System.getenv(varName));
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("UCAIP_PROJECT_ID");
+  }
+
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    originalPrintStream = System.out;
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown()
+      throws InterruptedException, ExecutionException, IOException, TimeoutException {
+    CancelBatchPredictionJobSample.cancelBatchPredictionJobSample(PROJECT, batchPredictionJobId);
+
+    // Assert
+    String cancelResponse = bout.toString();
+    assertThat(cancelResponse).contains("Cancelled the Batch Prediction Job");
+    TimeUnit.MINUTES.sleep(2);
+
+    // Delete the Batch Prediction Job
+    DeleteBatchPredictionJobSample.deleteBatchPredictionJobSample(PROJECT, batchPredictionJobId);
+
+    // Assert
+    String deleteResponse = bout.toString();
+    assertThat(deleteResponse).contains("Deleted Batch");
+    System.out.flush();
+    System.setOut(originalPrintStream);
+  }
+
+  @Test
+  public void testCreateBatchPredictionGeminiBigqueryJobSampleTest() throws IOException {
+    // Act
+    String batchPredictionDisplayName =
+        String.format(
+            "batch_prediction_gemini_bigquery_display_name_%s",
+            UUID.randomUUID().toString().replaceAll("-", "_").substring(0, 26));
+
+    CreateBatchPredictionGeminiBigqueryJobSample.createBatchPredictionGeminiBigqueryJobSample(
+        PROJECT,
+        batchPredictionDisplayName,
+        MODEL_ID,
+        "bigquery",
+        BIGQUERY_SOURCE_URI,
+        "bigquery",
+        BIGQUERY_DESTINATION_OUTPUT_URI_PREFIX);
+
+    // Assert
+    String got = bout.toString();
+    assertThat(got).contains(batchPredictionDisplayName);
+    assertThat(got).contains("response:");
+    batchPredictionJobId = got.split("Name: ")[1].split("batchPredictionJobs/")[1].split("\n")[0];
```

Comment:
Please use returned value instead of sysout for asserts.

Reply:
Done